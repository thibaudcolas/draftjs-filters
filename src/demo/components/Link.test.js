import React from "react"
import { shallow } from "enzyme"
import {
  EditorState,
  ContentState,
  convertFromHTML,
  convertFromRaw,
} from "draft-js"

import Link, { linkStrategy } from "./Link"

describe("Link", () => {
  it("renders", () => {
    const contentState = convertFromRaw({
      entityMap: {
        "0": {
          type: "LINK",
          mutability: "MUTABLE",
          data: {
            url: "www.example.com",
          },
        },
      },
      blocks: [
        {
          key: "6i47q",
          text: "NA link doc",
          type: "unstyled",
          depth: 0,
          inlineStyleRanges: [],
          entityRanges: [
            {
              offset: 3,
              length: 4,
              key: 0,
            },
          ],
          data: {},
        },
      ],
    })

    expect(
      shallow(
        <Link contentState={contentState} entityKey="1">
          Test
        </Link>,
      ),
    ).toMatchSnapshot()
  })
})

describe("linkStrategy", () => {
  it("works", () => {
    const editorState = EditorState.createWithContent(
      ContentState.createFromBlockArray(
        convertFromHTML(`<h1><a href="http://example.com/">Test</a></h1>`),
      ),
    )
    const currentContent = editorState.getCurrentContent()
    const callback = jest.fn()
    linkStrategy(currentContent.getFirstBlock(), callback, currentContent)
    expect(callback).toHaveBeenCalled()
  })
})
