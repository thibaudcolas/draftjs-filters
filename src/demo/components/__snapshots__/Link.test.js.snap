// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `
<span
  className="link"
  title="www.example.com"
>
  Test
</span>
`;
